site/content/blog/2025-10-07-solving-caido-labs/index.md

```diff
@@ -0,0 +1,430 @@
+
+![matchreplacer.png](images/matchreplacer.png)
+
+If we reload the page we notice that the admin panel is shown automatically.
+
+![matchrepadmin.png](images/matchrepadmin.png)
+
+With that the challenge is solved.
+
+![matchsolve.png](images/matchsolve.png)
+
+Notice that the Replacer rule stays active even if you close ZAP, so it's always good to disable the rules when you are done with them.
+
+## IDOR Vulnerability
+
+This challenge can be found on the next URL:
+
+[https://labs.cai.do/idor.php](https://labs.cai.do/idor.php)
+
+On this challenge we are shown the next site.
+
+![idorsite.png](images/idorsite.png)
+
+If we click the button the next information gets retrieved.
+
+![idorinfo.png](images/idorinfo.png)
+
+Checking the HTTP request we see that `user_id` is sent in a POST request.
+
+![idorreq.png](images/idorreq.png)
+
+We can send the request to Requester (Ctrl + W as shortcut) and try to send a different ID.
+
+![idoradmin.png](images/idoradmin.png)
+
+We got the information of the user Admin, but it tells us there is a kind of super admin user, to try to get it we send the request to the Fuzzer, and selecting the value of the user_id we add a payload of type "Numberzz" and set it to iterate from 1 to 50.
+
+![idorfuzz.png](images/idorfuzz.png)
+
+Once it's done the easier way to notice any difference is by changing the order by the size of the response Body, there we can see that besides the values "1" and "2", the value "42" is different.
+
+![idorfuzzed.png](images/idorfuzzed.png)
+
+Cheking that request we can verify that we found the ID of the super admin user.
+
+![idorsuper.png](images/idorsuper.png)
+
+While sorting by size works most of the time, there can be occasion where we want to look for something more specific, on that case we can write a "Fuzzer HTTP Processor" script:
+
+```js
+// Auxiliary variables/constants needed for processing.
+var count = 1;
+
+function processMessage(utils, message) {
+
+	//message.getRequestHeader().setHeader("X-Unique-Id", count);
+	count++;
+}
+
+function processResult(utils, fuzzResult){
+
+    var response = fuzzResult.getHttpMessage().getResponseBody().toString()
+	if (response.indexOf("super") !== -1)
+    {
+		fuzzResult.addCustomState("Key Custom State","Super found")
+	}
+	return true;
+}
+
+function getRequiredParamsNames(){
+	return [];
+}
+
+function getOptionalParamsNames(){
+	return [];
+}
+```
+
+All this script does is look for the word "super" on the response of the body and add a state that flags it, notice this is a very basic example and these scripts are really flexible, also if you want to know what every function does you can read the example script on ZAP since it is detailed there.
```

Comment:
```suggestion
All this script does is look for the word "super" in the response of the body and add a state that flags it, notice this is a very basic example and these scripts are really flexible, also if you want to know what every function does you can read the example script in ZAP since it is detailed there.
```